lib/src/__tests__/setup.js

```diff
@@ -36,3 +36,24 @@ console.error = (...args) => {
     throw new Error(message);
   }
 };
+
+// Convert any console warning into a thrown error
+const warn = console.warn;
+console.warn = (...args) => {
+  warn.apply(console, args);
```

Comment:
Probably we can move this logic up to the reusable function, and use it for both console.error and console.warn

Reply:
As you see fit, the module has a limited life expectancy.